src/compiler/utilities.ts

```diff
@@ -5337,7 +5350,25 @@ namespace ts {
         const kind = node.kind;
         return isStatementKindButNotDeclarationKind(kind)
             || isDeclarationStatementKind(kind)
-            || kind === SyntaxKind.Block;
+            || isBlockStatement(node);
+    }
+
+    /* @internal */
+    export function isStatementOrBlock(node: Node): node is Statement {
+        const kind = node.kind;
+        return isStatementKindButNotDeclarationKind(kind)
+            || isDeclarationStatementKind(kind)
+            || node.kind === SyntaxKind.Block;
+    }

```

Comment:
What about just having the only caller check `node.kind === SyntaxKind.Block || isStatement(node)`?  This function is kind of hard to explain as an independent entity.